### ISSUE TITLE
Todo: (clifford/geometric algebra, grassmann algebra, bivector, trivector, quad-vector, rotors)

### ISSUE THREAD

**Issue by recp** [recp/cglm#173]
* https://en.wikipedia.org/wiki/Bivector
* https://en.wikipedia.org/wiki/Multivector
* https://en.wikipedia.org/wiki/Bivector#Projective_geometry
* https://www.euclideanspace.com/maths/algebra/clifford/d4/functions/inverse/index.htm
* https://www.euclideanspace.com/maths/algebra/clifford/index.htm
* https://www.euclideanspace.com/maths/algebra/clifford/geometry/transforms/motors/index.htm

just TODO list, we don't have to implement all...

**Comment by Winter091** [comment]
It would be very cool to see some glm_ray_* functions, for example, ray-sphere collision test, ray-aabb collision test etc.

**Comment by recp** [comment]
@Winter091 agree, we already have `glm_ray_triangle()` but more about ray would be awesome, added TODOs

**Comment by Winter091** [comment]
@recp well, I won't be able to help with implementation, i'm pretty bad at linear algebra. I'm building simple minecraft clone, ray - aabb hit detection is needed there, so I had found [this blog](https://medium.com/@bromanz/another-view-on-the-classic-ray-aabb-intersection-algorithm-for-bvh-traversal-41125138b525). I copied unoptimized version to my project and it works pretty well, but at certain angles it fails to detect collision and I don't know why. Also I don't understand what `tmin` and `tmax` variable are doing there and what value they should be initially assigned to.

**Comment by recp** [comment]
@Winter091 no problem, I'll try to implement highly optimized versions (maybe with `SIMD`) of ray functions asap. Maybe someone else can implement them before me, it is possible. There are lot of projects I'm working on, you just need to wait a bit ;)

**Comment by crystalthoughts** [comment]
the Klein library is a good reference
